src/clr-angular/utils/drag-and-drop/draggable.spec.ts

```diff
@@ -6,32 +6,83 @@
 import {Component} from "@angular/core";
 import {TestBed} from "@angular/core/testing";
 import {By} from "@angular/platform-browser";
+import {NoopAnimationsModule} from "@angular/platform-browser/animations";
+
+import {ClrIconModule} from "../../icon/icon.module";
+import {DomAdapter} from "../dom-adapter/dom-adapter";
+import {ClrDragAndDropModule} from "./drag-and-drop.module";
 import {ClrDragHandle} from "./drag-handle";
 import {ClrDraggable} from "./draggable";
-import {ClrDragAndDropEventBus} from "./providers/drag-and-drop-event-bus";
 import {ClrDragEventListener} from "./providers/drag-event-listener";
+import {MOCK_DRAG_EVENT_LISTENER_PROVIDER} from "./providers/drag-event-listener.mock";
 import {ClrDragHandleRegistrar} from "./providers/drag-handle-registrar";
+import {ClrDraggableSnapshot} from "./providers/draggable-snapshot";
+import {ClrGlobalDragMode} from "./providers/global-drag-mode";
 
 export default function(): void {
     describe("ClrDraggable", function() {
-        describe("ClrDragHandle", function() {
+        describe("Basic Draggable", function() {
             beforeEach(function() {
-                TestBed.configureTestingModule({
-                    declarations: [NestedHandleTest, ClrDraggable, ClrDragHandle],
-                    providers: [ClrDragAndDropEventBus]
+                TestBed.configureTestingModule(
+                    {imports: [ClrDragAndDropModule, NoopAnimationsModule], declarations: [BasicDraggableTest]});
+                TestBed.overrideComponent(ClrDraggable, {
+                    set: {
+                        providers: [
+                            DomAdapter, ClrDragHandleRegistrar, ClrDraggableSnapshot, ClrGlobalDragMode,
+                            MOCK_DRAG_EVENT_LISTENER_PROVIDER
+                        ]
+                    }
                 });
 
-                this.fixture = TestBed.createComponent(NestedHandleTest);
+                this.fixture = TestBed.createComponent(BasicDraggableTest);
                 this.testComponent = this.fixture.componentInstance;
                 this.draggable = this.fixture.debugElement.query(By.directive(ClrDraggable));
                 this.dragEventListener = this.draggable.injector.get(ClrDragEventListener);
                 this.dragHandleRegistrar = this.draggable.injector.get(ClrDragHandleRegistrar);
+                this.globalDragMode = this.draggable.injector.get(ClrGlobalDragMode);
+                this.fixture.detectChanges();
             });
 
             afterEach(function() {
                 this.fixture.destroy();
             });
 
+            it("should have draggable class", function() {
+                expect(this.draggable.nativeElement.classList.contains("draggable")).toBeTruthy();
+            });
+
+            it("should emit event on drag start and have being-dragged class", function() {
```

Comment:
Shouldn't this be broken down into 2 tests? You are testing 2 different things here.

Reply:
Addressed this one